### PR TITLE
fix(issue): [Bug]: The "recover missing task plan → plan-slice" auto-dispatch rule re-dispatches `plan-slice` with no retry counter, allowing an uncounted regenerate loop

### LINKED ISSUE #1087: [Bug]: The "recover missing task plan → plan-slice" auto-dispatch rule re-dispatches `plan-slice` with no retry counter, allowing an uncounted regenerate loop
**Bug type:** Incorrect behavior or output

**Summary:** The auto-dispatch rule "executing → execute-task (recover missing task plan → plan-slice)" (`auto-dispatch.ts:1518-1600`) re-dispatches `plan-slice` to regenerate missing task-plan files with no per-unit retry counter and no rule-local circuit breaker, unlike its sibling rule "planning → plan-slice" which caps retries via `MAX_PRE_EXEC_RETRIES`.

**Steps to reproduce:**

1. In `/gsd auto`, reach `state.phase === "executing"` with an `activeTask` whose task-plan files are missing while the slice plan exists (and tasks are not embedded in the slice plan).
2. The rule dispatches `plan-slice` for `unitId = ${mid}/${sid}` (`auto-dispatch.ts:1585-1598`) with only a `debugLog` (`:1571`) — no counter increment, no cap.
3. If regeneration keeps failing to produce the task-plan files, the rule re-dispatches every cycle with no stop of its own.

**Expected behavior:** The recovery caps re-dispatch attempts per unit — mirroring the sibling's `MAX_PRE_EXEC_RETRIES = 2` keyed on `unitId` — and then returns an `action: "stop"` for manual intervention rather than re-dispatching indefinitely.

**Actual behavior:** The rule re-dispatches `plan-slice` with no retry counter; only the generic stuck-detector provides a backstop, not the rule itself.

**GSD version:** 1.3.0 (local HEAD build at commit `4d440d17`, ~2330 commits on HEAD)

**Node.js version:** v26.3.1

**Operating system:** macOS 26.5.1 (Darwin 25.5.0, arm64)

**Install method:** local checkout / locally compiled build

**Affected area:** Auto-mode/dispatch loop

**Logs, screenshots, and evidence:**

- `auto-dispatch.ts:1571-1598` — the recovery rule's dispatch path is `debugLog` (when debug enabled) followed by `return { action: "dispatch", unitType: "plan-slice", unitId:`${mid}/${sid}`, prompt: … }`, with no counter.
- `auto-dispatch.ts:1306-1322` — the counted sibling ("planning → plan-slice") does have a circuit breaker:

  ```
  const MAX_PRE_EXEC_RETRIES = 2;
  const retryCount = session.preExecRetryCount?.get(unitId) ?? 0;
  if (retryCount >= MAX_PRE_EXEC_RETRIES) {
    …
    return { action: "stop", reason: `Pre-execution checks failed … manual intervention required…`, level: "error", … };
  }
  ```

**Impact:** Wasted regeneration cycles and potential repeated dispatch on a unit that cannot self-repair, with no rule-local stop — the consistency gap carried forward from the missing-task-plan recovery family.

**Additional information:** In the recovery rule (`auto-dispatch.ts:1518-1600`), before dispatching `plan-slice`, add a per-unit streak counter keyed on `unitId = ${mid}/${sid}` (reuse `session.preExecRetryCount` or an analogous per-session map), increment it on each recovery dispatch, and once it reaches a small cap (mirror `MAX_PRE_EXEC_RETRIES = 2` from `auto-dispatch.ts:1309`) return an `action: "stop"` with a manual-intervention reason instead of re-dispatching. Reset/clear the counter for the unit once the task-plan files are found so a later legitimate recovery is not pre-tripped.

### Reproduction evidence

- **Repro status:** reproduced (static).
- **How tested:** Read the recovery rule body in full (`auto-dispatch.ts:1518-1600`) and confirmed it contains no counter increment or cap — only a `debugLog` and the `plan-slice` dispatch. Contrasted with the sibling "planning → plan-slice" rule (`auto-dispatch.ts:1306-1322`) which maintains `session.preExecRetryCount` and stops at `MAX_PRE_EXEC_RETRIES`.
- **Raw result:** recovery rule: 0 counter references; sibling: `MAX_PRE_EXEC_RETRIES = 2` + `session.preExecRetryCount` get/delete.
- **Confidence & corrections:** High. All cited `file:line` re-confirmed at HEAD `4d440d17`.

### PR DESCRIPTION
**Pull request by jeremymcs**
## Summary
- Added bounded missing task-plan recovery retries with regression tests; whitespace verification passed, while test execution was blocked by unavailable npm registry access.

## Verification
- Completed in the repository worktree before push.

## Related Issue
- Closes #1087
- [#1087 [Bug]: The "recover missing task plan → plan-slice" auto-dispatch rule re-dispatches `plan-slice` with no retry counter, allowing an uncounted regenerate loop](https://github.com/open-gsd/gsd-pi/issues/1087)

## User
- @astarktc

## Repo
- `open-gsd/gsd-pi`

## Branch
- `issue/1087-bug-the-recover-missing-task-plan-plan-s-1782847045`

---

> [!NOTE]
> **Low Risk**
> Localized auto-dispatch circuit-breaker logic mirroring existing pre-exec retries; regression tests cover the new paths with no auth or data-handling impact.
> 
> **Overview**
> Fixes **#1087** by bounding the **executing → recover missing task plan → plan-slice** auto-dispatch path, which could re-dispatch `plan-slice` forever when per-task `PLAN` files never appeared.
> 
> **`AutoSession`** now tracks **`missingTaskPlanRetryCount`** per slice (`M###/S##`), separate from **`preExecRetryCount`**, and clears it on session **`reset()`**. On each recovery dispatch the counter increments; after **2** failed attempts auto-mode **stops** with an error asking for manual fixes. When task plans exist again, the counter is cleared so a later recovery can retry without inheriting stale counts.
> 
> Regression tests use **legacy** `milestones/…/tasks/` layouts to cover increment, exhaustion, and counter clearing without touching pre-exec retries.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 2fb17d2ed84c9ea8cab210e15f7748ee0b7b08a3. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

---
<a href="https://app.blacksmith.sh/open-gsd/codesmith/gsd-pi/pr/1101"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is enabled.</sup>

### CODE REVIEW

**Comment by jeremymcs** [comment]
✅ **Verification — all 1 task is addressed**

Re-checked PR #1101 against [#1087 [Bug]: The "recover missing task plan → plan-slice" auto-dispatch rule re-dispatches `plan-slice` with no retry counter, allowing an uncounted regenerate loop](https://github.com/open-gsd/gsd-pi/issues/1087).

## Subtasks
- [x] **[Bug]: The "recover missing task plan → plan-slice" auto-dispatch rule re-dispatches `plan-slice` with no retry counter,**

_Re-run from Patchdeck to refresh._

File: src/resources/extensions/gsd/auto-dispatch.ts [hunk 1]
```diff
@@ -1598,6 +1610,7 @@ export const DISPATCH_RULES: DispatchRule[] = [
         };
       }
 
+      session?.preExecRetryCount?.delete(unitId);
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/auto-dispatch.ts]: **Accepted** — queued for a code change.
**In progress** — applying the accepted fix.
**Verifying** — checking the applied changes.
**Resolved** — addressed in commit `2fb17d2`.

**Comment by jeremymcs** [comment]
Started an automated PR update.

Accepted review feedback is being applied.
Progress and final results will be posted in the related review thread(s).

File: src/resources/extensions/gsd/auto-dispatch.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/auto-dispatch.ts]: Addressed in commit `2fb17d2`.
